rasa_sdk/forms.py

```diff
@@ -689,3 +689,44 @@ def _get_entity_type_of_slot_to_fill(
                 return None
 
         return entity_type
+
+
+class FormSlotsValidatorAction(Action):
+    """An action that validates if every extracted slot is valid."""
+
+    def name(self) -> Text:
+        """Unique identifier of this action."""
+
+        raise NotImplementedError("An action must implement a name")
+
+    async def run(
+        self,
+        dispatcher: "CollectingDispatcher",
+        tracker: "Tracker",
+        domain: "DomainDict",
+    ) -> List[EventType]:
+        """Execute the side effects of this action.
+
+        Args:
+            dispatcher: the dispatcher which is used to
+                send messages back to the user.
+            tracker: the state tracker for the current user.
+            domain: the bot's domain.
+        Returns:
+            A dictionary of ``rasa_sdk.events.Event`` instances.
+        """
+
+        slots_to_validate: Dict[Text, Any] = tracker.form_slots_to_validate()
+        validation_events = []
+
+        for slot_name, slot_value in slots_to_validate.items():
+            function_name = f"validate_{slot_name}"
+            fn = getattr(self, function_name)
+            if fn(tracker, domain, slot_value):
```

Comment:
I think we should keep this in line to the behavior in the old deprecated `FormAction` (see [here](https://github.com/RasaHQ/rasa-sdk/blob/2fff067d26b64aaabaa5ce60163081ffe0418891/rasa_sdk/forms.py#L408)). This would also allow users to simply copy over their validate functions.

The signature should imo look like:

```python
def validate_my_slot(
        self,
        slot_dict: Dict[Text, Any],
        dispatcher: "CollectingDispatcher",
        tracker: "Tracker",
        domain: "DomainDict",
    ) -> List[EventType]
```

Reply:
Not sure why you pasted this signature while the previous signature was clearly different :)
In addition, it seems like having `slot_dict` as an argument is useless because you can easily get slots from `tracker`.

Reply:
sorry, overlooked the `slot_dict`. You're right, should be the extracted slot value instead.